### PR TITLE
Add --restricted_roles option to Monitors API

### PR DESCRIPTION
**Pull request by pierreln-dd**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue, or add one feature, at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

This PR adds the "restricted_roles" argument to the Monitors API endpoint (see https://docs.datadoghq.com/api/latest/monitors/#create-a-monitor)

### Description of the Change

Added the argument --restricted_roles to all the necessary parsers in order to be able to create/update/validate monitors with restricted roles

### Verification Process

I installed the project locally and performed some local tests against my account. I specifically checked that I could add multiple roles, and checked in my DD account that the roles were properly added.
I also ran the tox integration tests but had a lot of errors that were not from my change.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](https://github.com/DataDog/datadogpy/blob/master/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

File: datadog/dogshell/monitor.py [hunk 1]
```diff
@@ -175,6 +184,11 @@ def _post(cls, args):
             "message": args.message,
             "options": options
         }
+        if args.restricted_roles:
+            restricted_roles = sorted(set([rr.strip() for rr in args.restricted_roles.split(",") if rr.strip()]))
+            body["restricted_roles"] = restricted_roles
+        else:
+            restricted_roles = None
```
Comment (skarimo) [inline datadog/dogshell/monitor.py:191]: Why is this needed? Doesnt seem to do anything

File: datadog/dogshell/monitor.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def setup_parser(cls, subparsers):
         post_parser.add_argument(
             "--message", help="message to include with notifications" " for this monitor", default=None
         )
+        post_parser.add_argument(
+            "--restricted_roles", help="list of unique role identifiers allowed to edit the monitor", default=None
```
Comment (skarimo) [inline datadog/dogshell/monitor.py:40]:
```suggestion
            "--restricted_roles", help="comma-separated list of unique role identifiers allowed to edit the monitor", default=None
```

File: datadog/dogshell/monitor.py (hunk 1 above)
Comment (pierreln-dd) [reply datadog/dogshell/monitor.py]: Removed in new commit

File: datadog/dogshell/monitor.py (hunk 2 above)
Comment (pierreln-dd) [reply datadog/dogshell/monitor.py]: Good suggestion, thanks

File: datadog/dogshell/monitor.py [hunk 3]
```diff
@@ -414,14 +446,24 @@ def _validate(cls, args):
         options = None
         if args.options is not None:
             options = json.loads(args.options)
-
         if args.tags:
```
Comment (skarimo) [inline datadog/dogshell/monitor.py:449]:
```suggestion

        if args.tags:
```

**Review by skarimo — APPROVED** [review]
